### DISCUSSION TITLE
Prompt tracing with easy to use UI/UX

### DISCUSSION THREAD

**Discussion by vicperdana** [microsoft/semantic-kernel#9301] [Q&A] [answered]
Hi team,

I’m looking for solutions that build on SK's telemetry framework to help non-technical users easily debug and refine prompts and their outputs. Ideally, this would provide a simple interface for non-technical users to view interactions, identify issues, and modify prompts without coding.

Key requirements:

- Telemetry-based for tracking prompt performance (response times, total token used breakdown for prompt and output)
- Easy-to-use for non-technical users to navigate, drill, expand specific prompts and their responses.
- Preferably SDK in C# / .NET 

Explored the following solutions but unfortunately they don't support .NET SDK. 
[LangSmith](https://www.langchain.com/langsmith)
[PromptFlow](https://learn.microsoft.com/en-us/azure/ai-studio/how-to/monitor-quality-safety?tabs=azure-studio#consume-monitoring-results)

The latest [blog](https://devblogs.microsoft.com/semantic-kernel/step-by-step-guide-to-building-a-powerful-ai-monitoring-dashboard-with-semantic-kernel-and-azure-monitor/) is heading in the right direction but it doesn't seem to have capability to drill/navigate to particular prompts without serious Kusto skills!

Any guidance is appreciated.

**Comment by sophialagerkranspandey** [comment, marked as answer]
Hi @vicperdana, thanks for the question. Please feel free to ping me as we would love to chat further. We recommend using Prompty through plugins for semantic kernel, Aspire for infrastructure development.
